### PR TITLE
Add tests for sphinx documentation

### PR DESCRIPTION
**Pull request by tiran**
* Add ``tox -e doc`` target to test docs
* Add spell checker and word list
* Fix spelling errors
* Fix some markup

- [ ] review spelling list for errors

### CODE REVIEW

**Comment by encukou** [comment]
Thanks!
I have some more urgent doc work lined up (moving `python-ldap.org` to RTD). Let me put this PR on hold until that's done.

**Comment by tiran** [comment]
I had to remove the link checker because one site doesn't work properly:

```
Warning, treated as error:
/home/heimes/dev/redhat/python-ldap/Doc/installing.rst.rst:63:broken link: https://www.lfd.uci.edu/~gohlke/pythonlibs/ (('Connection aborted.', OSError(0, 'Error')))
```

File: Doc/spelling_wordlist.txt [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+cb
+cfg
+changeNumber
+changesOnly
+changeType
+changeTypes
+cidict
+clientctrls
+conf
+Continously
```
Comment (encukou) [inline Doc/spelling_wordlist.txt]: This is a typo, should be Contin<strong>u</strong>ously

**Comment by encukou** [comment]
I'll go through the wordlist and push a commit, rather than commenting individually.

**Comment by tiran** [comment]
I found another issue: behavio**u**r.

**Comment by encukou** [comment]
“behaviour” is fine in British English.

**Pull request merged**